### PR TITLE
Add missing feature switch

### PR DESCRIPTION
**Pull request by JosephWoodward**

### CODE REVIEW

File: src/Microsoft.FeatureManagement/FeatureManagementOptions.cs [hunk 1]
```diff
@@ -13,5 +13,11 @@ public class FeatureManagementOptions
         /// If missing feature filters are not ignored an exception will be thrown when attempting to evaluate a feature that depends on a missing feature filter.
         /// </summary>
         public bool IgnoreMissingFeatureFilters { get; set; }
+
+        /// <summary>
+        /// Controls the behavior of feature evaluation when the dependent feature declaration is missing.
```
Comment (jimmyca15) [inline src/Microsoft.FeatureManagement/FeatureManagementOptions.cs:18]: Controls the behavior of feature evaluation when the specified feature does not exist.
If missing features are not ignored, an exception will be thrown when attempting to evaluate them.

File: src/Microsoft.FeatureManagement/FeatureManagementError.cs [hunk 2]
```diff
@@ -16,6 +16,11 @@ public enum FeatureManagementError
         /// <summary>
         /// A feature filter configured for the feature being evaluated is an ambiguous reference to multiple registered feature filters.
         /// </summary>
-        AmbiguousFeatureFilter
+        AmbiguousFeatureFilter,
+
+        /// <summary>
+        /// A feature filter is missing configuration
```
Comment (jimmyca15) [inline src/Microsoft.FeatureManagement/FeatureManagementError.cs:22]: The specified feature does not exist.

File: src/Microsoft.FeatureManagement/FeatureManager.cs [hunk 3]
```diff
@@ -140,6 +140,9 @@ private async Task<bool> IsEnabledAsync<TContext>(string feature, TContext appCo
                         }
                     }
                 }
+            } else if (!_options.IgnoreMissingFeatures)
```
Comment (jimmyca15) [inline src/Microsoft.FeatureManagement/FeatureManager.cs:143]: New line

**Comment by jimmyca15** [comment]
@zhenlan thoughts on changing the default behavior to throw on a missing feature? I would prefer that we do it that way. It is a breaking behavioral change, but not a breaking API change. Therefore it wouldn't require a major version bump.

**Comment by jimmyca15** [comment]
Discussed with @zhenlan, let's go ahead and make the default behavior to not throw for a missing feature. There is a scenario for this where a developer codes ahead of someone who is setting up configuration.

**Comment by JosephWoodward** [comment]
@jimmyca15 Thanks for the clarification, I'll update the PR.

**Comment by JosephWoodward** [comment]
@jimmyca15 Updated following your comments and now ready for review.

File: src/Microsoft.FeatureManagement/FeatureManagementOptions.cs [hunk 4]
```diff
@@ -13,5 +13,11 @@ public class FeatureManagementOptions
         /// If missing feature filters are not ignored an exception will be thrown when attempting to evaluate a feature that depends on a missing feature filter.
         /// </summary>
         public bool IgnoreMissingFeatureFilters { get; set; }
+
+        /// <summary>
+        /// Controls the behavior of feature evaluation when the specified feature does not exist.
+        /// If missing features are not ignored, an exception will be thrown when attempting to evaluate them.
+        /// </summary>
+        public bool IgnoreMissingFeatures { get; set; }
```
Comment (zhenlan) [inline src/Microsoft.FeatureManagement/FeatureManagementOptions.cs:21]: I think we want this defaults to `true`.

File: tests/Tests.FeatureManagement/FeatureManagement.cs [hunk 5]
```diff
@@ -483,5 +491,59 @@ public async Task SwallowsExceptionForMissingFeatureFilter()
...
+        public async Task SwallowsOnMissingFeature()
+        {
+            IConfiguration config = new ConfigurationBuilder().AddJsonFile("appsettings.json").Build();
+
+            var services = new ServiceCollection();
+
+            services
+                .Configure<FeatureManagementOptions>(options =>
+                {
+                    options.IgnoreMissingFeatures = true;
```
Comment (zhenlan) [inline tests/Tests.FeatureManagement/FeatureManagement.cs:532]: I would remove this so we can test the default behavior and ensure no regressions.

File: tests/Tests.FeatureManagement/FeatureManagement.cs [hunk 6]
```diff
@@ -33,6 +35,12 @@ public async Task ReadsConfiguration()
 
             var services = new ServiceCollection();
 
+            services
+                .Configure<FeatureManagementOptions>(options =>
+                {
+                    options.IgnoreMissingFeatures = true;
```
Comment (zhenlan) [inline tests/Tests.FeatureManagement/FeatureManagement.cs:41]: Why is this change needed?

File: tests/Tests.FeatureManagement/FeatureManagement.cs [hunk 7]
```diff
@@ -14,6 +14,7 @@
 using System.Linq;
 using System.Net;
 using System.Net.Http;
+using System.Reflection;
```
Comment (zhenlan) [inline tests/Tests.FeatureManagement/FeatureManagement.cs:17]: Sorry, I may miss it as it's not very easy to tell. What we are using in the reflection namespace?

File: src/Microsoft.FeatureManagement/FeatureManager.cs [hunk 8]
```diff
@@ -141,6 +141,10 @@ private async Task<bool> IsEnabledAsync<TContext>(string feature, TContext appCo
                     }
                 }
             }
+            else if (!_options.IgnoreMissingFeatures)
+            {
+                throw new FeatureManagementException(FeatureManagementError.MissingFeature, $"The feature declaration for specified feature '{feature}' was not found.");
+            }
```
Comment (zhenlan) [inline src/Microsoft.FeatureManagement/FeatureManager.cs:147]: We should log a warning even if `IgnoreMissingFeatures` is `true`. Suggest change below
```cs
else
{
    string errorMessage = $"The feature declaration for specified feature '{feature}' was not found.";
    if (!_options.IgnoreMissingFeatures)
    {
        throw new FeatureManagementException(FeatureManagementError.MissingFeature, errorMessage);
    }
    else
    {
        _logger.LogWarning(errorMessage);
    }
}
```

**Comment by jimmyca15** [comment]
This PR has not had any recent activity. In an effort to address #68, a new PR, #78, has been created. Any future discussion can be moved there.
